token-contracts/WorkerToken/src/main/java/finance/omm/score/WorkerTokenImpl.java

```diff
@@ -0,0 +1,159 @@
+package finance.omm.score;
+
+import static java.math.BigInteger.TEN;
+import static java.math.BigInteger.ZERO;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import finance.omm.core.score.interfaces.WorkerToken;
+import finance.omm.utils.db.EnumerableSet;
+import score.Address;
+import score.ArrayDB;
+import score.Context;
+import score.DictDB;
+import score.VarDB;
+import score.annotation.EventLog;
+import score.annotation.External;
+import score.annotation.Optional;
+
+public class WorkerTokenImpl implements WorkerToken {
+
+    private static final String TAG = "Worker Token";
+
+    private static final String BALANCES = "balances";
+    private static final String TOTAL_SUPPLY = "total_supply";
+    private static final String DECIMALS = "decimals";
+    private static final String WALLETS = "wallets";
+    public static final String WORKER_TOKEN_SYMBOL = "OMMWT";
+
+    private VarDB<BigInteger> totalSupply = Context.newVarDB(TOTAL_SUPPLY, BigInteger.class);
+    private VarDB<BigInteger> decimals = Context.newVarDB(DECIMALS, BigInteger.class);
+    private DictDB<Address, BigInteger> balances = Context.newDictDB(BALANCES, BigInteger.class);
+    private EnumerableSet<Address> wallets = new EnumerableSet<>(WALLETS, Address.class);
+
+    public WorkerTokenImpl(BigInteger _initialSupply, BigInteger _decimals, @Optional boolean _update) {
+        if (_update) {
+            onUpdate();
+            return;
+        }
+
+        Context.println(TAG + "| on install event:" + Context.getAddress());
+        if (_initialSupply == null || _initialSupply.compareTo(ZERO) < 0) {
+            Context.revert("Initial supply cannot be less than zero");
+        }
+
+        if (_decimals == null || _decimals.compareTo(ZERO) < 0) {
+            Context.revert("Decimals cannot be less than zero");
+        }
+
+        BigInteger totalSupply = _initialSupply.multiply(pow(TEN, _decimals.intValue()));
+        Context.println(TAG + "| total_supply " + totalSupply);
+
+        this.totalSupply.set(totalSupply);
+        this.decimals.set(_decimals);
+        this.balances.set(Context.getCaller(), totalSupply);
```

Comment:
Though this would be the case if we had to deploy Java Score from the start, these balances are already set for python scores and should not be updated when we try to migrate python score to Java.

Reply:
that is why update parameter is provided, if this contract is deployed using _update = true, the code block that is remarked will not be executed. (line 36)

Reply:
Sorry my bad! I seem to have missed the return statement on update check.